linode/terraform-provider-linode
# new: Add `linode_child_account` and `linode_child_accounts` data sources; add fields to `linode_account` data source

## 📝 Description

This pull request adds two new data sources: `linode_child_account` and `linode_child_accounts`. These can be used to query information about child accounts under the current Linode account.

Additionally, this pull request adds two new fields to the `linode_account` schema:
- `capabilities` - The capabilities of the account.
- `active_since` - When the account was activated.

**NOTE: Billing-related fields have been intentionally excluded from the `linode_account` schema.**

## ✔️ How to Test

The following tests assume you have pulled down this PR locally and are using a `LINODE_TOKEN` and `LINODE_URL` an account with children.

### Integration Tests

```
export TEST_TAGS="parent_child"

make int-test PKG_NAME=linode/account
make int-test PKG_NAME=linode/childaccount
make int-test PKG_NAME=linode/childaccounts
```

### Unit Tests

```
make unit-test
```

### Manual Testing

1. In a Terraform provider sandbox environment, apply the following configuration:

```terraform

data "linode_child_accounts" "all" {}

data "linode_child_accounts" "filter" {
  filter {
    name = "email"
    values = [data.linode_child_accounts.all.child_accounts.0.email]
  }
}

data "linode_child_account" "single" {
  euuid = data.linode_child_accounts.all.child_accounts.0.euuid
}

output "all" {
  value = data.linode_child_accounts.all.child_accounts
}

output "filtered" {
  value = data.linode_child_accounts.filter.child_accounts
}

output "single" {
  value = data.linode_child_account.single
}
``` 
2. Ensure all outputs are populated as expected.
